tests/unit/resources/test_datastores.py

```diff
@@ -173,6 +173,20 @@ def test_resize(self, mock_get, mock_post):
         self.assertEqual(datastore.data, resource_data[0])
         mock_post.assert_called_once_with('/datastores/12345/resize', {'size': 2048}, custom_headers=None)
 
+    @mock.patch.object(Connection, "post")
+    @mock.patch.object(Connection, "get")
+    def test_set_policy(self, mock_get, mock_post):
```

Comment:
we will need two unit tests.

test_set_policy ( with policy object )
test_set_policy_with_name ( with policy name )

Reply:
Added couple of unit testcases